core/src/main/java/org/stellar/anchor/sep31/Sep31Service.java

```diff
@@ -539,21 +539,53 @@ private void validateSenderAndReceiver()
       throw new BadRequestException("receiver_id cannot be empty.");
     }
 
-    // TODO: Populate customer type with the first receiver type; this is a temporary fix for cases
-    // where customerType is required
     Sep12Operation sep12Operation = Context.get().getAsset().getSep31().getSep12();
-    String receiverType = null;
+    Sep12GetCustomerResponse receiver;
+
+    // Fix for https://github.com/stellar/java-stellar-anchor-sdk/issues/717 (ANCHOR-149)
+    // TODO: check if there is a better way to do it, i.e. request a type first for a customer OR
+    // stop iteration after first hit.
     if (sep12Operation != null) {
-      Optional<String> receiverTypeOptional =
-          sep12Operation.getReceiver().getTypes().keySet().stream().findFirst();
-      receiverType = receiverTypeOptional.orElse(null);
-    }
-    Sep12GetCustomerRequest request =
-        Sep12GetCustomerRequest.builder().id(receiverId).type(receiverType).build();
-    Sep12GetCustomerResponse receiver = this.customerIntegration.getCustomer(request);
-    if (receiver == null || receiver.getStatus() != Sep12Status.ACCEPTED) {
-      infoF("Customer (receiver) info needed for request ({})", Context.get().getRequest());
-      throw new Sep31CustomerInfoNeededException("sep31-receiver");
+      Set<String> types = sep12Operation.getReceiver().getTypes().keySet();
+      List<Sep12GetCustomerRequest> requests =
+          types.stream()
+              .map(type -> Sep12GetCustomerRequest.builder().id(receiverId).type(type).build())
+              .collect(Collectors.toList());
+      List<Sep12GetCustomerResponse> responses = new ArrayList<>();
+
+      for (Sep12GetCustomerRequest request : requests) {
+        try {
+          responses.add(this.customerIntegration.getCustomer(request));
+        } catch (AnchorException e) {
+          infoF("Failed to get customer (receiver) info with an error ({})", e);
+        }
+      }
+
+      List<Sep12GetCustomerResponse> accepted =
+          responses.stream()
+              .filter(Objects::nonNull)
+              .filter(x -> x.getStatus() == Sep12Status.ACCEPTED)
+              .collect(Collectors.toList());
+
+      if (accepted.isEmpty()) {
+        infoF("Customer (receiver) info needed for request ({})", Context.get().getRequest());
+        throw new Sep31CustomerInfoNeededException(types.toString());
+      }
+      if (accepted.size() != 1) {
+        infoF(
+            "Ambiguous customer info for request ({}) having multiple accepted responses ({})",
+            Context.get().getRequest(),
+            accepted);
+        throw new Sep31AmbiguousCustomerInfoException(accepted.toString());
+      }
```

Comment:
I'm struggling with that we should do in this scenario. The Anchor Platform v1 has the assumption that types are mutually exclusive, so if this happens, the AP should probably return 500 to the Sending Anchor, right?

I don't see any changes to the Anchor Platform in this PR, only the underlying Java SDK. Where is the logic for how the platform handles these AnchorExceptions?

Reply:
@JakeUrban 
org.stellar.anchor.reference.controller.GlobalControllerExceptionHandler

Reply:
Ok, it looks like its throwing 500 for all exceptions that aren't specifically cased in the file. I don't see `AnchorException` listed, so I'm assuming 500 is returned?

Reply:
Yes. There are handlers for specific AnchorExceptions, but generic one returns 500